tests/testsuite/update.rs

```diff
@@ -2616,3 +2616,39 @@ fn update_breaking_mixed_pinning_renaming() {
             "#]],
     );
 }
+
+#[cargo_test]
+fn update_breaking_pre_release() {
```

Comment:
From <https://github.com/rust-lang/cargo/pull/14250/files#r1680348122>:

> The `update --breaking` had filtered the `prerelease` version to update to. So here equals to no upgrade to prerelease also. :/

Since that is the case, we might want to have UI tests for

1. `2.0.0-beta.22` is available, but `2.0.0-beta.21` won't upgrade to it.
2. `2.0.0` is available, but `2.0.0-beta.21` won't upgrade to it.
2. `3.0.0` is available, and `2.0.0-beta.21` will upgrade to it.

We should document the first two as an unresolved issue. That is something we want to support, especially the second one.

Reply:
Added `update_breaking_pre_release_upgrade` to reflect the current behavior.

> 2.0.0 is available, but 2.0.0-beta.21 won't upgrade to it.

This case out of my expection, it seems need to support.

Reply:
Task added: <https://github.com/rust-lang/cargo/issues/12425#issuecomment-2186198258>.